lib/smart_proxy_dynflow_core/api.rb

```diff
@@ -1,11 +1,22 @@
 require 'sinatra/base'
 require 'multi_json'
 
+# rubocop:disable Lint/HandleExceptions
+begin
+  require 'proxy/log'
+  require 'proxy/helpers'
+  require 'sinatra/authorization'
+rescue LoadError
+end
+# rubocop:enable Lint/HandleExceptions
+
 module SmartProxyDynflowCore
   class Api < ::Sinatra::Base
     TASK_UPDATE_REGEXP_PATH = %r{/tasks/(\S+)/(update|done)}
     helpers Helpers
 
+    include ::Sinatra::Authorization::Helpers unless Settings.instance.standalone
```

Comment:
We also have this:
https://github.com/theforeman/smart-proxy/blob/6c0ee87d874fd10bb1757538182b8332a1c96833/lib/smart_proxy_main.rb#L49

That makes me think you can use `helpers :Sinatra::Authorization` or may even be already included in base.

Btw, I do think we should write real Rack middleware to handle auth. I submitted my WIP to https://github.com/theforeman/smart-proxy/pull/789 just so it's not only on my desktop.

For the short term: is `Settings.instance` initialized when this Api class is loaded? Should it be `if defined?(::Sinatra::Authorization::Helpers)`?

Reply:
I'm ok with include. Looks like it's how it's commonly used in the Smart Proxy itself as well.